src/sync/index.js

```diff
@@ -403,111 +408,126 @@ async function sync(db){
+    try {
+      value = await contract.call('getTotalVotes', { methodArgs: [], senderAddress: senderAddress});
+    } catch(err){
+      console.error(`Error: getTotalVotes for oracle ${oracleAddress}, ${err.message}`);
+      resolve();
+      return;
+    }
+  }
+
+  let balances = _.map(value[0].slice(0, oracle.options.length), (balance_BN) => {
+    return balance_BN.toJSON();
   });
+
+  try {
+    await db.Oracles.updateOne({address: oracleAddress}, { $set: { amounts: balances }});
+    resolve();
+  } catch(err){
+    console.error(`Error: update oracle ${oracleAddress}, ${err.message}`);
+    resolve();
+  }
 }
 
-function updateTopicBalance(topicAddress, db, resolve){
-  db.Topics.findOne({address: topicAddress}).then(function(topic){
-    if(!topic){
+async function updateTopicBalance(topicAddress, db, resolve){
+  var topic;
+  try{
+    topic = await db.Topics.findOne({address: topicAddress});
+    if (!topic){
+      console.error(`Error: find 0 topic ${topicAddress} in db to update`);
       resolve();
       return;
     }
-    const contract = qclient.Contract(topicAddress, Contracts.TopicEvent.abi);
-    contract.call('getTotalBets', { methodArgs: [], senderAddress: senderAddress})
-      .then(
-        (value)=>{
-          let balances = _.map(value[0].slice(0, topic.options.length), (balance_BN) =>{
-            return balance_BN.toJSON();
-          });
-          db.Topics.updateOne({address: topicAddress}, { $set: { qtumAmount: balances } }, function(err, res){
-            if(err){
-              console.error(`Error: ${err.message}`);
-            }
-            resolve();
-          });
-      });
+  } catch(err){
+    console.error(`Error: find topic ${topicAddress} in db, ${err.message}`);
+    resolve();
+    return;
+  }
 
-    contract.call('getTotalVotes', { methodArgs: [], senderAddress: senderAddress})
-      .then(
-        (value)=>{
-          let balances = _.map(value[0].slice(0, topic.options.length), (balance_BN) =>{
-            return balance_BN.toJSON();
-          });
-          db.Topics.updateOne({address: topicAddress}, { $set: { botAmount: balances } }, function(err, res){
-            if(err){
-              console.error(`Error: ${err.message}`);
-            }
-            resolve();
-          });
-      });
+  const contract = new Qweb3Contract(QTUM_RPC_ADDRESS, topicAddress, Contracts.TopicEvent.abi);
+  var totalBetsValue, totalVotesValue;
+  try{
+    // TODO(frankobe): mk this two async
+    totalBetsValue = await contract.call('getTotalBets', { methodArgs: [], senderAddress: senderAddress});
+    totalVotesValue = await contract.call('getTotalVotes', { methodArgs: [], senderAddress: senderAddress});
+  }catch(err){
+    console.error(`Error: getTotalBets for topic ${topicAddress}, ${err.message}`);
+    resolve();
+    return;
+  }
+
+  let totalBetsBalances = _.map(totalBetsValue[0].slice(0, topic.options.length), (balance_BN) =>{
+    return balance_BN.toJSON();
   });
+
```

Comment:
totalBetsValue and totalBetsValue[0] need undefined check
totalVotesValue is the same.